### DISCUSSION TITLE
Workflow erro 401

### DISCUSSION THREAD

**Discussion by oGrandeK** [balta-io/forum#285] [Tira Dúvidas] [answered]
Olá novamente professor, e sim é mais um erro com o workflow rs. Eu só peço ajuda aqui porque eu procuro em todo canto e não acho uma solução.

> Run dotnet nuget push "bin/Release/*.nupkg" -k *** -s https://nuget.pkg.github.com/oGrandeK/index.json --skip-duplicate
  
> Pushing ae5a8daf47[6](https://github.com/oGrandeK/my-custom-private-package/actions/runs/4430685843/jobs/7772732186#step:6:6)074f8[6](https://github.com/oGrandeK/my-custom-private-package/actions/runs/4430685843/jobs/7772732186#step:6:7)b62f5ded72a610adMyCustomPrivatePackage.1.0.0.nupkg to 'https://nuget.pkg.github.com/oGrandeK'...
  PUT https://nuget.pkg.github.com/oGrandeK/
warn : Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.
  Unauthorized https://nuget.pkg.github.com/oGrandeK/ 141ms
error: Response status code does not indicate success: 401 (Unauthorized).

**Comment by andrebaltieri** [comment, marked as answer]
bom dia, @oGrandeK, fica tranquilo... normal acontecer esses erros...
Dá uma olhada neste link que tem as soluções:
👉 https://github.com/balta-io/forum/discussions?discussions_q=403

**Reply by oGrandeK** [reply]
Olá novamente professor, fucei a internet novamente e descobri que eu tinha que criar um TOKEN no meu perfil do github nas SETTINGS -> DEVELOPER SETTINGS.

Feito isso eu fui no meu yml e troquei o `${{ secrets.GITHUB_TOKEN }}` por meu token recém criado no DEVELOPER SETTINGS e funcionou, quando eu tentava usar o GITHUB_TOKEN ele estava dando erro. O senhor sabe me explicar o porque dava erro de autenticação quando eu usava ${{ secrets.GITHUB_TOKEN }} ??

Consegue me explicar como devo escrever no yml pra ter acesso a esse token sem precisar digitalo puro no yml? 

![image](https://user-images.githubusercontent.com/89154726/225787971-1e832d34-146d-492b-aac7-3f5bcc64ab3f.png)

**Reply by andrebaltieri** [reply]
Boa tarde, @oGrandeK show demais!!!!
Qual erro está ocorrendo agora?

**Reply by oGrandeK** [reply]
Boa noite, não aconteceu nenhum erro agora não professor. Vou tentar explicar de forma mais lúdica.

Precisei criar um `TOKEN` dentro do meu perfil do GitHub( caminho: `SETTINGS -> DEVELOPER SETTINGS -> PERSONAL ACCESS TOKENS -> TOKENS (CLASSIC)` para poder rodar o `Publish` do worflow que até então estava dando `error 401 Unauthorized`.

Acontece que dentro do `workflow.yml` se eu deixasse o parâmetro `-k` da seguinte forma: `-k ${{ secrets.GITHUB_TOKEN }}` o `Publish` sempre retornava esse erro `401`. Quando eu alterei o parâmetro `-k` para `-k <MEUTOKENCRIADONOCAMINHOACIMA>` o `Publish` acabou funcionando.

Minhas dúvidas eram simplesmente o porque desse erro `401` aparecer e como eu poderia fazer para não precisar digitar `<MEUTOKENCRIADONOCAMINHOACIMA>` dentro do `workflow.yml` até porque precisei subir meu `workflow.yml` com esse meu `TOKEN` pro repositório remoto onde todo mundo tem acesso a esse `TOKEN` kkkkkkk.

Espero que eu tenha conseguido explicar minhas dúvidas para o senhor.

Abraços e obrigado professor <3

**Reply by oGrandeK** [reply]
Uma outra dúvida me surgiu, será que esse erro 401 Unauthorized no Publish se dá pelo fato de eu estar upando o código no meu repositório pessoal em vez de uma organização?

**Reply by andrebaltieri** [reply]
@oGrandeK exatamente. Se você tivesse em uma organização, com os devidos privilégios, seria mais simples.
